### PR TITLE
Add sdk attribute to reporting API (refs GH-2599)

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by mattrobenolt** [comment]
Will the client be responsible for sending this new interface? or would we derive from User-Agent?

**Comment by mattrobenolt** [comment]
Sorry, out of `sentry_client`, not User-Agent.

**Comment by dcramer** [comment]
@mattrobenolt we could probably derive as a followup, but the client would be responsible (and we could remove sentry_client from header probably)

**Comment by dcramer** [comment]
screenshot 2016-04-12 09 55 43

File: src/sentry/web/api.py [hunk 1]
```diff
@@ -335,6 +335,11 @@ def process(self, request, project, auth, helper, data, **kwargs):
         # mutates data
         data = helper.validate_data(project, data)
 
+        if 'sdk' not in data:
```
Comment (mattrobenolt) [inline src/sentry/web/api.py]: :+1:

File: src/sentry/coreapi.py [hunk 2]
```diff
@@ -328,6 +328,21 @@ def _process_fingerprint(self, data):
...
         return result
 
+    def parse_client_as_sdk(self, value):
+        if not value:
+            return
+        try:
+            name, version = value.split('/', 1)
+        except ValueError:
+            try:
+                name, verison = value.split(' ', 1)
```
Comment (mattrobenolt) [inline src/sentry/coreapi.py]: `verison`

Comment (mattrobenolt) [inline src/sentry/coreapi.py]: Also, why is space a valid delimiter here? Is it allowed to be `raven-js 2.0.0`? instead of `raven-js/2.0.0`?

**Comment by mattrobenolt** [comment]
lgtm

File: src/sentry/coreapi.py (hunk 2 above)
Comment (dcramer) [inline src/sentry/coreapi.py]: we dont enforce anything

**Pull request merged**